litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/ToolBar/index.tsx

```diff
@@ -10,114 +10,108 @@ import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+  handleRefreshRateChange: (refreshRate: number) => void;
+  handleForceUpdate: () => void;
+}
+
+const ToolBar: React.FC<ToolBarProps> = ({
+  timeRange,
+  refreshInterval,
+  handleRangeChange,
+  handleRefreshRateChange,
+  handleForceUpdate,
+}) => {
   const classes = useStyles();
   const outlinedInputClasses = useOutlinedInputStyles();
   const { t } = useTranslation();
-  const dashboard = useActions(DashboardActions);
-  const selectedDashboard = useSelector(
-    (state: RootState) => state.selectDashboard
-  );
-
   const dateRangeSelectorRef = React.useRef<HTMLDivElement>(null);
   const [isDateRangeSelectorPopoverOpen, setDateRangeSelectorPopoverOpen] =
     React.useState(false);
-  const [refreshRate, setRefreshRate] = React.useState<number>(
-    selectedDashboard.refreshRate ? selectedDashboard.refreshRate : 0
-  );
+  const [range, setRange] = React.useState<RangeType>(timeRange);
+  const [refreshRate, setRefreshRate] = React.useState<number>(refreshInterval);
   const [openRefresh, setOpenRefresh] = React.useState(false);
-  const handleCloseRefresh = () => {
-    setOpenRefresh(false);
-  };
-
-  const handleOpenRefresh = () => {
-    setOpenRefresh(true);
-  };
-
-  const clearTimeOuts = async () => {
-    let id = window.setTimeout(() => {}, 0);
-    while (id--) {
-      window.clearTimeout(id);
-    }
-
-    return Promise.resolve(id === 0);
-  };
+  const handleCloseRefresh = () => setOpenRefresh(false);
+  const handleOpenRefresh = () => setOpenRefresh(true);
 
   const CallbackFromRangeSelector = (
     selectedStartDate: string,
     selectedEndDate: string
   ) => {
-    const startDateFormatted: string = moment(selectedStartDate).format();
-    const endDateFormatted: string = moment(selectedEndDate)
-      .add(23, 'hours')
-      .add(59, 'minutes')
-      .add(59, 'seconds')
-      .format();
-    dashboard.selectDashboard({
-      range: { startDate: startDateFormatted, endDate: endDateFormatted },
-    });
-    const endDate: number =
-      new Date(moment(endDateFormatted).format()).getTime() / 1000;
-    const now: number = Math.round(new Date().getTime() / 1000);
-    const diff: number = Math.abs(now - endDate);
-    const maxLim: number =
-      (selectedDashboard.refreshRate ?? DEFAULT_REFRESH_RATE) / 1000 !== 0
-        ? (selectedDashboard.refreshRate ?? DEFAULT_REFRESH_RATE) / 1000 +
-          MINIMUM_TOLERANCE_LIMIT
-        : DEFAULT_TOLERANCE_LIMIT;
+    const startDateFormatted: string = `${
+      new Date(moment(selectedStartDate).format()).getTime() / 1000
+    }`;
+    const endDateFormatted: string = `${
+      new Date(
+        moment(selectedEndDate)
+          .add(23, 'hours')
+          .add(59, 'minutes')
+          .add(59, 'seconds')
+          .format()
+      ).getTime() / 1000
+    }`;
+    setRange({ startDate: startDateFormatted, endDate: endDateFormatted });
+    // relativeTime to be sent from date range selector
+    // if (relativeTime === INVALID_RELATIVE_TIME_RANGE) {
```

Comment:
unused comments can be removed

Reply:
This time range picker is a dev in progress feature so we have kept it as a dev comment.